www/public/repl/repl.js

```diff
@@ -165,11 +165,37 @@ async function processInputQueue() {
 // Callbacks to JS from Rust
 // ----------------------------------------------------------------------------
 
+var ROC_PANIC_INFO = null;
+
+function send_panic_msg_to_js(rocstr_ptr, tag_id) {
+    const { memory } = repl.app.exports;
+    const memoryBuffer = new Uint8Array(memory.buffer);
+
+    const rocStrBytes = new Uint8Array(memory.buffer, rocstr_ptr, 12);
+
+    const strPtr = rocStrBytes[0] | (rocStrBytes[1] << 8) | (rocStrBytes[2] << 16) | (rocStrBytes[3] << 24);
+    const strLen = rocStrBytes[4] | (rocStrBytes[5] << 8) | (rocStrBytes[6] << 16) | (rocStrBytes[7] << 24);
+
+    const stringBytes = new Uint8Array(memory.buffer, strPtr, strLen);
+    const decoder = new TextDecoder('utf-8');
```

Comment:
should this re-use the `repl.textDecoder` instead of creating a new one?

Reply:
Yes